src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDocumentGenerationService.kt

```diff
@@ -59,24 +65,36 @@ class PdfDocumentGenerationService(@Autowired private val gotenbergApi: Gotenber
     }.build()
 }
 
-interface DocumentDetail<T> {
-  val name: String
+interface Data<T> {
   fun data(): T
 }
 
-data class ClassPathDocumentDetail(override val name: String, val path: String = "/templates/images/$name") :
-  DocumentDetail<MultipartInputStreamFileResource> {
-  override fun data() = MultipartInputStreamFileResource(ClassPathResource(path).inputStream, name)
+interface ImageData<T> : Data<T> {
+  val name: String
+}
+
+data class ClassPathImageData(override val name: String) :
```

Comment:
I'm going to stop worrying about it but this `name` property is actually a `filename`; whereas the `filename` property of `class MultipartInputStreamFileResource` is ...?  Right now I can't see why we still need that.  Must be time for me tea.

Reply:
it's also the name of the image in the HTML, so needs to match the HTML.  Might do as you suggested yesterday and try and and some sort of type to encapsulate the Image name so we can reuse in the HTML generation and the PDF generation, so this might disappear